### PR TITLE
Wrap and indent help descriptions for options and commands

### PR DESCRIPTION
**Pull request by shadowspawn**
Finish off #956 with Jest tests and last fixes. See #956 for description and discussion.

(I realised when adding tests for this I have never tried the somewhat undocumented argsDescription parameter to `.description`, and not currently being tested here.)

### CODE REVIEW

**Comment by shadowspawn** [comment]
FYI: @Ephigenia

**Comment by shadowspawn** [comment]
No rush @abetomo, I requested review here accidentally. Thanks for other reviews already today!

File: index.js [hunk 1]
```diff
@@ -1329,6 +1340,48 @@ function pad(str, width) {
...
+}
+
+/**
+ * Optionally wrap the given str to a max width of width characters per line
+ * while indenting with indent spaces. Do not wrap the text when the minWidth
+ * is not hit.
+ *
+ * @param {String} str
+ * @param {Number} width
+ * @param {Number} indent
```
Comment (abetomo) [inline index.js]: The document parameter is missing `minWidth`.

File: index.js [hunk 2]
```diff
@@ -1329,6 +1340,48 @@ function pad(str, width) {
...
+ * @return {String}
+ * @api private
+ */
+function wrap(str, width, indent) {
+  var regex = new RegExp('.{1,' + (width - 1) + '}([\\s\u200B]|$)|[^\\s\u200B]+?([\\s\u200B]|$)', 'g');
+  var lines = str.match(regex) || [];
+  var result = lines.map(function(line, i) {
```
Comment (abetomo) [inline index.js]: How about returning directly without assigning to a variable?

File: index.js (hunk 1 above)
Comment (shadowspawn) [reply index.js]: Prompted by this, I moved minWidth to an internal detail.

File: index.js (hunk 2 above)
Comment (shadowspawn) [reply index.js]: Done

**Pull request merged**